src/lamp_py/tableau/jobs/rt_rail.py

```diff
@@ -73,6 +73,63 @@ def __init__(self) -> None:
+            "       SELECT "
+            "           canon_trips.direction_id "
+            "           , CASE WHEN canon_trips.trunk_route_id = 'Green' THEN canon_trips.route_id ELSE canon_trips.trunk_route_id END as route_id "
+            "           , static_stops.parent_station "
+            "           , ROW_NUMBER() OVER (PARTITION BY canon_trips.static_version_key,canon_trips.direction_id,canon_trips.route_id ORDER BY static_stop_times.stop_sequence) as stop_sequence "
+            "           , canon_trips.static_version_key "
+            "       FROM "
+            "           ( "
+            "           SELECT "
+            "               DISTINCT ON (coalesce(static_trips.branch_route_id,static_trips.trunk_route_id),static_route_patterns.direction_id,static_route_patterns.static_version_key) "
+            "               static_route_patterns.direction_id as direction_id "
+            "               , static_route_patterns.representative_trip_id as representative_trip_id "
+            "               , static_trips.trunk_route_id as trunk_route_id "
+            "               , coalesce(static_trips.branch_route_id, static_trips.trunk_route_id) as route_id "
+            "               , static_route_patterns.static_version_key as static_version_key "
+            "           FROM "
+            "               static_route_patterns "
+            "           JOIN static_trips on "
+            "               static_route_patterns.representative_trip_id = static_trips.trip_id "
+            "               AND static_route_patterns.static_version_key = static_trips.static_version_key "
+            "           JOIN static_routes on "
+            "               static_routes.route_id = static_trips.route_id "
+            "               AND static_routes.static_version_key = static_trips.static_version_key "
+            "           WHERE "
+            "               static_routes.route_type < 2 "
+            "               AND (static_route_patterns.route_pattern_typicality = 1 "
+            "                   or static_route_patterns.route_pattern_typicality = 5) "
+            "           order by "
+            "               coalesce(static_trips.branch_route_id, static_trips.trunk_route_id), "
+            "               static_route_patterns.direction_id, "
+            "               static_route_patterns.static_version_key, "
+            "               static_route_patterns.route_pattern_typicality desc) as canon_trips "
+            "       JOIN static_stop_times on "
+            "           canon_trips.representative_trip_id = static_stop_times.trip_id "
+            "           AND canon_trips.static_version_key = static_stop_times.static_version_key "
+            "       JOIN static_stops on "
+            "           static_stop_times.stop_id = static_stops.stop_id "
+            "           AND static_stop_times.static_version_key = static_stops.static_version_key ) as drop_station "
```

Comment:
sanity check for my sake. 

this gets every station on every trip, then its filtered out to only be the first station for every trip.

we use that for the filtering later rather than the `ve.canonical_stop_sequence`

Reply:
Yes, this subquery is the same one used to create `canonical_stop_sequence` but we are limiting it to the first `parent_station` in each each direction and then doing an anti-join with the `drop_flag` field.